### DISCUSSION TITLE
Debug cells?

### DISCUSSION THREAD

**Discussion by Michael-F-Ellis** [janpfeifer/gonb#96] [Q&A]
Is is possible to run the debugger on a cell.  I'm working in VsCode with the latest Jupyter extension.

**Comment by janpfeifer** [comment]
Unfortunately not, Jupyter has a special debugging protocol that I haven't implemented yet.

I suspect it won't be trivial, and I foresee some busy weeks ahead in my day job, so it will likely take at least a few weeks (I have no idea yet if its even possible ...)

I created issue #97 to follow up on it.

**Reply by Michael-F-Ellis** [reply]
Thanks!  I read your starting point link to the Jupyter Debugger doc and, yeah, it does look like a serious undertaking, especially given that your kernel is compiling each cell behind the scenes.

Maybe an interim solution, at least for VsCode, would be a feature that would wrap the cell's code in a Go Test function and export it to a file. VsCode knows about test func signatures and conveniently puts a little "Debug Test" button above each func in a test file. I'm guessing that other major IDE's have similar capabilities.

I did it by hand today when a cell in a notebook used some code that was doing type assertions on maps with interface{} values.  I could probably be happy forever with a little automation wrapped around the process of creating and populating the test file and function.

Just a thought...
